scripts/zones/Windurst_Woods/npcs/Ponono.lua

```diff
@@ -11,34 +14,58 @@ require("scripts/globals/status")
 
 function onTrade(player,npc,trade)
     local newRank = tradeTestItem(player,npc,trade,tpz.skill.CLOTHCRAFT)
-
+    local moralmanifest = player:getQuestStatus(OTHER_AREAS_LOG,tpz.quest.id.otherAreas.A_MORAL_MANIFEST)
+    local tradeGil = trade:getGil()
     if newRank ~= 0 then
         player:setSkillRank(tpz.skill.CLOTHCRAFT,newRank)
         player:startEvent(10012,0,0,0,0,newRank)
     end
+    if moralmanifest == QUEST_ACCEPTED and player:getCharVar("moral") == 2 then
+        if (trade:hasItemQty(828,1) and trade:hasItemQty (830, 1) and tradeGil == 10000) then -- Trade Velvet Cloth, Rainbow Cloth and 10k
+            player:tradeComplete()
```

Comment:
I think we are trying to get everyone to start using the **npcUtil.tradeHas** method of checking items now if possible, to avoid issues,
A reveiwer can elaborate more on this if possible, but the function is found in the global npc_util.lua,
you should be able to search other scripts using this style of trading, again i may be incorrect, i am just pointing out potential issues for reviewers to look through, dont feel this is an issue unless other have asked to have this changed.